### PR TITLE
[Boron] Workaround for SARA R4 ppp session getting broken

### PR DESCRIPTION
**Pull request by avtolstoy**
### Problem

1. PPP link gets broken on LTE Borons and SoMs when sending larget packets:
```
0000040423 [lwip] TRACE: pppos_netif_output[3]: proto=0x21, len = 40
0000040426 [hal] TRACE: Outputing 1320 bytes
0000040433 [lwip] TRACE: pppos_netif_output[3]: proto=0x21, len = 1316
0000042175 [hal] TRACE: Outputing 16 bytes
0000042175 [lwip] TRACE: pppos_write[3]: len=12
0000042360 [hal] TRACE: Outputing 22 bytes
0000042360 [lwip] TRACE: pppos_write[3]: len=18
0000043426 [hal] TRACE: Outputing 1321 bytes
0000043427 [lwip] TRACE: pppos_netif_output[3]: proto=0x21, len = 1316
0000047176 [hal] TRACE: Outputing 16 bytes
0000047176 [lwip] TRACE: pppos_write[3]: len=12
0000048361 [hal] TRACE: Outputing 22 bytes
0000048361 [lwip] TRACE: pppos_write[3]: len=18
0000049429 [hal] TRACE: Outputing 1321 bytes
0000049430 [lwip] TRACE: pppos_netif_output[3]: proto=0x21, len = 1316
0000052177 [hal] TRACE: Outputing 16 bytes
0000052177 [lwip] TRACE: pppos_write[3]: len=12
0000054362 [hal] TRACE: Outputing 22 bytes
0000054362 [lwip] TRACE: pppos_write[3]: len=18
0000057178 [hal] TRACE: Outputing 16 bytes
0000057178 [lwip] TRACE: pppos_write[3]: len=12
0000060363 [hal] TRACE: Outputing 22 bytes
0000060363 [lwip] TRACE: pppos_write[3]: len=18
0000061432 [hal] TRACE: Outputing 1321 bytes
0000061433 [lwip] TRACE: pppos_netif_output[3]: proto=0x21, len = 1316
0000062179 [hal] TRACE: Outputing 16 bytes
0000062179 [lwip] TRACE: pppos_write[3]: len=12
0000066364 [hal] TRACE: Outputing 22 bytes
0000066364 [lwip] TRACE: pppos_write[3]: len=18
0000067180 [hal] TRACE: Outputing 16 bytes
0000067180 [lwip] TRACE: pppos_write[3]: len=12
0000072181 [hal] TRACE: Outputing 16 bytes
0000072181 [lwip] TRACE: pppos_write[3]: len=12
0000072365 [lwip] TRACE: IPV6CP: timeout sending Config-Requests
0000077182 [hal] TRACE: Outputing 16 bytes
0000077182 [lwip] TRACE: pppos_write[3]: len=12
0000082183 [hal] TRACE: Outputing 16 bytes
0000082183 [lwip] TRACE: pppos_write[3]: len=12
0000085436 [hal] TRACE: Outputing 1321 bytes
0000085437 [lwip] TRACE: pppos_netif_output[3]: proto=0x21, len = 1316
0000087184 [hal] TRACE: Outputing 16 bytes
0000087184 [lwip] TRACE: pppos_write[3]: len=12
0000092185 [lwip] TRACE: No response to 10 echo-requests
0000092185 [lwip] TRACE: Serial link appears to be disconnected.
0000092186 [lwip] TRACE: ppp phase changed[3]: phase=11
```

### Solution

1. Decrease MTU on LTE Borons and SoMs to a lower value (`NCP_SARA_R410_MTU`) **TODO: for now it's at 1300, should be changed to the one that actually works after the testing**

### Steps to Test

1. Can be tested with `MDM_01_socket_writes_with_length_more_than_1023_work_correctly`

### Example App

N/A

### References

N/A

---

### Completeness

- [x] User is totes amazing for contributing!
- [x] Contributor has signed CLA ([Info here](https://github.com/spark/firmware/blob/develop/CONTRIBUTING.md))
- [x] Problem and Solution clearly stated
- [x] Run unit/integration/application tests on device
- [ ] Added documentation
- [ ] Added to CHANGELOG.md after merging (add links to docs and issues)

---

- [Bugfix] [Boron] Workaround for SARA R4 ppp session getting broken and system power manager fix [#1726](https://github.com/particle-iot/device-os/pull/1726)
- [Bugfix] Fixes system power manager re-enabling charging every 1s with a battery connected (now every 60s) [#1726](https://github.com/particle-iot/device-os/pull/1726)

### CODE REVIEW

File: hal/network/lwip/ublox/pppncpnetif.cpp [hunk 1]
```diff
@@ -55,6 +55,8 @@ enum class NetifEvent {
     PowerOn = 5
 };
 
+unsigned NCP_SARA_R410_MTU = 1300;
```
Comment (technobly) [inline hal/network/lwip/ublox/pppncpnetif.cpp]: Currently 1012 passes, 1013 fails.

File: user/build.mk [hunk 2]
```diff
@@ -71,7 +71,7 @@ endif
 INCLUDE_DIRS += $(MODULE_PATH)/libraries
 
 CFLAGS += -DSPARK_PLATFORM_NET=$(PLATFORM_NET)
-CPPFLAGS += -std=gnu++11
+CPPFLAGS += -std=gnu++14
```
Comment (technobly) [inline user/build.mk]: C++ 11 to C++ 14.  This seems like too large of a change to be subtly included in this PR.  Is this necessary to switch to now and should we be doing it in it's own PR?

Comment (avtolstoy) [reply user/build.mk]: This shoudln't cause any issues. C++14 is backwards compatible with C++11 and simply enables additional C++14 features.

Comment (m-mcgowan) [reply user/build.mk]: what is it we need it for? Can we achieve the same in C++11? I can appreciate it's good to upgrade, but I would take the language upgrade at the same time as we upgrade the compiler.

Comment (avtolstoy) [reply user/build.mk]: `std::make_unique` mainly, as it was forgotten in C++11. FYI we are already enabling C++14 in `services`, `hal` and `system` and gcc 5.3.1 fully supports it (see https://gcc.gnu.org/projects/cxx-status.html#cxx14). There shouldn't be any side-effects or overhead of simply enabling C++14 support in user apps.

I've heard the concerns about this change a couple of times already, but no legitimate reasons against apart from 'what if'.

File: hal/src/boron/network/sara_ncp_client.cpp [hunk 3]
```diff
@@ -559,7 +559,7 @@ int SaraNcpClient::waitReady() {
         // Disable voltage translator
         modemSetUartState(false);
         // Hard reset the modem
-        modemHardReset();
+        modemHardReset(true);
```
Comment (technobly) [inline hal/src/boron/network/sara_ncp_client.cpp]: What is the optimization here?  I see we now if initialization of the modem fails, we hard reset the modem which results in it turning off, and we don't turn it back on for R410.  Was there a double turn on occurring and now there's only one?  Any chance the turn on will not re-occur?
